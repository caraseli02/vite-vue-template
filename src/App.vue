<script setup lang="ts">
import { computed } from 'vue'
import MainNav from '~/components/Navigation/MainNav.vue'
import AppHeader from '~/components/Navigation/AppHeader.vue'
import AppSidebar from '~/components/AppSidebar.vue'
// import skeletonLoade from "~/components/skeletonLoader.vue";

// import { useStore } from "vuex";
// const store = useStore();

// const user = computed(() => store.getters["auth/authUser"]);
const user = computed(() => {
  return {
    name: 'Juan',
    surname: 'Perez',
    email: '',
    id: '1',
    workplace: '1',
  }
})
</script>

<template>
  <div class="min-w-screen min-h-screen box-border">
    <AppHeader />
    <AppSidebar />
    <main :key="user?.id" class="max-w-md mx-auto px-4" :class="user ? 'lg:pl-24' : ''">
      <router-view />
    </main>
    <!-- <Footer /> -->
    <section
      v-if="user"
      class="glass-blue fixed bottom-0 w-full rounded-none rounded-t-lg lg:w-20 lg:h-full lg:rounded-r-lg lg:rounded-t-none"
    >
      <MainNav />
    </section>
  </div>
</template>

<style>
@import "./assets/css/icons.css";
@import url("https://fonts.googleapis.com/css2?family=Poppins:ital,wght@0,400;0,500;0,700;1,100;1,400&display=swap");
/* @import "assets/style.css"; */

* {
  font-family: "Poppins";
}

html {
  background: radial-gradient(circle at top left, #dad4cd 0%, #00c6fb 20%, #005bea 100%);
}

.dark {
  background: radial-gradient(circle at top right, #014393 0%, #020c6b 20%, #010948 100%);
}

.slide-fade-enter-active {
  transition: all 0.3s ease;
}

.slide-fade-leave-active {
  transition: all 0.8s cubic-bezier(1, 0.5, 0.8, 1);
}

.slide-fade-enter-from,
.slide-fade-leave-to {
  transform: translateY(10px);
  opacity: 0;
}
</style>
