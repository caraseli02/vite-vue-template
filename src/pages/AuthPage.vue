<template>
  <section class="max-w-sm mx-auto mt-10">
    <div class="mb-4">
      <h1 class="text-3xl leading-9 font-extrabold text-primary mb-4">
        APIMOSA APP
      </h1>

      <PrimaryBtn v-show="!isLogin" id="authLogin" @click="setIsLogin(true)">
        INICIAR SESIÃ“N
        <LoginIcon class="ml-2 w-5 h-5" />
      </PrimaryBtn>

      <PrimaryBtn v-show="isLogin" id="authCreate" @click="setIsLogin(false)">
        CREAR CUENTA
        <IdentificationIcon class="ml-2 w-5 h-5" />
      </PrimaryBtn>
    </div>
    <Transition name="list">
      <SignIn v-if="isLogin" />
      <Register v-else />
    </Transition>
  </section>
</template>

<script setup lang="ts">
  import SignIn from '~/components/auth/SignIn.vue'
  import PrimaryBtn from '~/components/buttons/PrimaryBtn.vue'
  import Register from '~/components/auth/SignUp.vue'
  import { ref } from 'vue'
  import { IdentificationIcon, LoginIcon } from '@heroicons/vue/outline'

  const props = defineProps({
    showSignUp: {
      type: Boolean,
      default: false
    }
  })

  const isLogin = ref(!props.showSignUp)

  const setIsLogin = (nextVal: boolean) => {
    isLogin.value = nextVal
  }
</script>

<style scoped>
  .list-enter-active,
  .list-leave-active {
    transition: all 1s ease;
  }

  .list-enter-from {
    opacity: 0;
    transform: translateX(-50%);
    z-index: 1;
  }

  .list-leave-to {
    position: absolute;
    opacity: 0;
    transform: translateX(50%);
    z-index: -1;
  }
</style>
