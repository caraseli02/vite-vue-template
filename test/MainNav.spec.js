import { render, fireEvent } from '@testing-library/vue'
import VueRouter from 'vue-router'
import { expect } from 'vitest'
import { createRouter, createWebHistory } from 'vue-router'

import App from '@/App.vue'
import HomePage from '@/pages/HomePage.vue'
import AuthPage from '@/pages/AuthPage.vue'
import DashboardPage from '@/pages/DashboardPage.vue'
import AttendsList from '@/pages/AttendsList.vue'
import UserPerfil from '@/pages/UserPerfil.vue'

const routes = [
  {
    path: '/',
    component: HomePage
  },
  {
    path: '/auth',
    component: AuthPage
  },
  {
    path: '/dashboard',
    component: DashboardPage
  },
  {
    path: '/perfil',
    component: UserPerfil
  },
  {
    path: '/info',
    component: AttendsList
  }
]

test('full app rendering/navigating', async () => {
  // Notice how we pass a `routes` object to our render function.
  const { getByTestId } = render(App, { routes })

  expect(getByTestId('location-display')).toHaveTextContent('/')

  await fireEvent.click(getByTestId('about-link'))

  expect(getByTestId('location-display')).toHaveTextContent('/about')
})

test('mount MainNav component', async () => {
  expect(MainNav).toBeTruthy()

  const wrapper = mount(MainNav, {
    global: {
      plugins: [router]
    }
  })
  // console.log(wrapper.html());
  //find a tag with href /auth
  expect(wrapper.find('a[href="/dashboard"]').exists()).toBe(true)
  expect(wrapper.find('a[href="/info"]').exists()).toBe(true)
  expect(wrapper.find('a[href="/"]').exists()).toBe(true)

  // check if $route.path === link.path
  expect(wrapper.vm.$route.path).toBe('/')

  await wrapper.find('a[href="/"]').trigger('click')
  expect(wrapper.html()).toMatchSnapshot()
})
