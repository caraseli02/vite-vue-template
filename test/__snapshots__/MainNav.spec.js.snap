// Vitest Snapshot v1

exports[`mount MainNav component 1`] = `
"<div class=\\"flex lg:flex-col items-center justify-center h-full space-x-4 lg:space-x-0 lg:space-y-4 py-2 lg:py-4 lg:rouded-xl\\"><a href=\\"/\\" class=\\"group\\">
    <div class=\\"transform -translate-y-8 lg:-translate-y-0 lg:translate-x-8 duration-300 w-36 flex justify-center items-center h-10 lg:w-14 lg:h-14 rounded-xl bg-secondary text-primary transform shadow-xl\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" fill=\\"none\\" viewBox=\\"0 0 24 24\\" stroke-width=\\"2\\" stroke=\\"currentColor\\" aria-hidden=\\"true\\" class=\\"w-6 h-6\\">
        <path stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" d=\\"M3 12l2-2m0 0l7-7 7 7M5 10v10a1 1 0 001 1h3m10-11l2 2m-2-2v10a1 1 0 01-1 1h-3m-6 0a1 1 0 001-1v-4a1 1 0 011-1h2a1 1 0 011 1v4a1 1 0 001 1m-6 0h6\\" />
      </svg><span class=\\"ml-2\\">Inicio</span></div>
  </a><a href=\\"/dashboard\\" class=\\"group\\">
    <div class=\\"opacity-50 border-2 border-blue-500 duration-300 w-10 flex justify-center items-center h-10 lg:w-14 lg:h-14 rounded-xl bg-secondary text-primary transform shadow-xl\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" fill=\\"none\\" viewBox=\\"0 0 24 24\\" stroke-width=\\"2\\" stroke=\\"currentColor\\" aria-hidden=\\"true\\" class=\\"w-6 h-6\\">
        <path stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" d=\\"M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z\\" />
      </svg><span class=\\"ml-2\\" style=\\"display: none;\\">Dashboard</span></div>
  </a><a href=\\"/info\\" class=\\"group\\">
    <div class=\\"opacity-50 border-2 border-blue-500 duration-300 w-10 flex justify-center items-center h-10 lg:w-14 lg:h-14 rounded-xl bg-secondary text-primary transform shadow-xl\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" fill=\\"none\\" viewBox=\\"0 0 24 24\\" stroke-width=\\"2\\" stroke=\\"currentColor\\" aria-hidden=\\"true\\" class=\\"w-6 h-6\\">
        <path stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" d=\\"M9 19v-6a2 2 0 00-2-2H5a2 2 0 00-2 2v6a2 2 0 002 2h2a2 2 0 002-2zm0 0V9a2 2 0 012-2h2a2 2 0 012 2v10m-6 0a2 2 0 002 2h2a2 2 0 002-2m0 0V5a2 2 0 012-2h2a2 2 0 012 2v14a2 2 0 01-2 2h-2a2 2 0 01-2-2z\\" />
      </svg><span class=\\"ml-2\\" style=\\"display: none;\\">Info</span></div>
  </a></div>"
`;
